src/GHCMod.hs

```diff
@@ -333,7 +342,14 @@ main = handler $ do
 
 progMain :: (Options,[String]) -> IO ()
 progMain (globalOptions,cmdArgs) = do
-    (res,_) <- runGhcModT globalOptions $ ghcCommands cmdArgs
+    let
```

Comment:
Since `setMMappedFiles` is part of the exposed API I think it would be better to always load the mappings, i.e. move this somewhere inside runGhcModT or even runGmlTWith, since the mappings aren't used unless the ghc session is active anyways.

Unless I'm misunderstanding something here. I don't really get why you'd think it's a good idea to "mutate" the read-only option environment here, that's usually a good indicator for a dirty hack :p